packages/flutter/lib/src/services/flutter_version.dart

```diff
@@ -15,7 +15,17 @@
 abstract final class FlutterVersion {
   const FlutterVersion._();
 
-  /// The Flutter version used to compile the app.
+  /// The value is a versioning string representing the version of Flutter
+  /// used to compile the app, possibly followed by whitespace and other
+  /// version and build details.
+  ///
+  /// Flutter doesn't follow [semantic versioning](https://semver.org/).
+  /// See [here](https://docs.flutter.dev/release/breaking-changes)
```

Comment:
Alternatively, I could link to https://github.com/flutter/flutter/blob/master/docs/releases/Release-versioning.md or copy that text to the docs here. What do you think?